code/modules/research/nanites/nanite_programs/weapon.dm

```diff
@@ -87,11 +87,7 @@
 	addtimer(CALLBACK(src, .proc/boom), clamp((nanites.nanite_volume * 0.35), 25, 150))
 
 /datum/nanite_program/explosive/proc/boom()
-	var/nanite_amount = nanites.nanite_volume
-	var/dev_range = FLOOR(nanite_amount/200, 1) - 1
-	var/heavy_range = FLOOR(nanite_amount/100, 1) - 1
-	var/light_range = FLOOR(nanite_amount/50, 1) - 1
-	explosion(host_mob, dev_range, heavy_range, light_range)
+	dyn_explosion(get_turf(host_mob), nanites.nanite_volume / 50)
```

Comment:
Does this change behavior?

Reply:
It should make it scale more smoothly with volume, like chemical explosions do. Following the guidelines in the documentation, this scaling should cap at roughly the same powerlevel as before at 500 nanites, but there might be some differences inbetween.
The more important part of this change is that it should handle scaling beyond 500 better than using the previous system.